codegenerator/cli/src/rescript_types.rs

```diff
@@ -197,19 +216,20 @@ impl RescriptTypeExpr {
         }
     }
 
-    pub fn to_rescript_schema(&self) -> String {
+    pub fn to_rescript_schema(&self, type_name: &String) -> String {
         match self {
             Self::Identifier(type_ident) => type_ident.to_rescript_schema(),
             Self::Variant(items) => {
                 let item_schemas = items
                     .iter()
                     .map(|item| {
                         format!(
-                            r#"S.object(s =>
+                            r#"S.object((s): {} =>
 {{
   s.tag("case", "{}")
   {}({{payload: s.field("payload", {})}})
 }})"#,
+                            type_name,
```

Comment:
I think named parameters might be a bit easier to read with these longer string interpolations:

https://doc.rust-lang.org/std/fmt/#named-parameters

Also you don't have to pass the same value in twice like item.name here